### ISSUE TITLE
Client crash when drawing emojis

### ISSUE THREAD

**Issue by rdelian** [citizenfx/fivem#1572]
# Info
Since a lot of my users started to crash recently with _'some gfx errors'_ and I didn't update the gamemode in a while, I started to investigate it.

The trigger seems to be Drawing on screen with natives, especially when I draw Emojis + Chars, but `GetTextScreenLineCount` / `EndTextCommandGetWidth` has to be called too.

```lua
-- CitizenFX_log
Exception in unsafe stubbed GFx hooks: c0000005 at 00007ff730943c63.
*client_crash*
```

If you need crash dumps or any aditional informations, let me know.

# Repro
https://github.com/rdelian/repro_gfx_err/blob/main/client.lua
It takes a few seconds before the crash, sometimes it just fails to draw anything on screen and it will throw a client script error

**Comment by blattersturm** [comment]
This might be a repro for [this report from the forums](https://forum.cfx.re/t/crash-ntdll-dll-rtlpwaitoncriticalsection-0xb1/4875165/3), yes.

**Comment by rdelian** [comment]
I will provide a few dumps in the next 15 minutes or so

**Comment by blattersturm** [comment]
I somehow can't currently repro it on _debug_ builds at least when waiting 'a few seconds' (at ~143 FPS), but the mention of measuring text (on MainThrd) while drawing similar text (on Render) makes it look like a race condition somewhere.

**Comment by rdelian** [comment]
[CfxCrashDump_2022_08_14_13_41_42.zip](https://github.com/citizenfx/fivem/files/9333467/CfxCrashDump_2022_08_14_13_41_42.zip)
[CfxCrashDump_2022_08_14_13_45_16.zip](https://github.com/citizenfx/fivem/files/9333468/CfxCrashDump_2022_08_14_13_45_16.zip)

A way to instant crash (_for me atleast_) is to put [`GetTextWidth(TEXT, 0.3)` ](https://github.com/rdelian/repro_gfx_err/blob/main/client.lua#L26) in a for loop

**Comment by blattersturm** [comment]
It _does_ seem to correctly crash on retail builds, with a crash in the GFx heap allocator (after showing some broken emoji for a frame).

This might be somewhat related to 103e46dc947a4bd3a6276ba418a74c8d639fc60b, or it might be a different way the original issue represents itself.

**Comment by blattersturm** [comment]
Peculiar bonus is how a locally-build retail-style build *also* seems immune to this. Is this another case of having inadvertently fixed it?!

**Comment by blattersturm** [comment]
This turned out to be an anticheat hook having a _curious_ side effect. Fixed in `fivem-private@0e12b5487213ad821d8d70741fee7bbe0abd5a1b`.
